practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/validation/RoleAssignmentValidation.java

```diff
@@ -32,12 +36,18 @@ public ValidationResponse validate(RoleAssignment entity) {
         if (isNullOrBlank(entity.getRoleId())) {
             return new ValidationResponse(false, "roleId must have a value");
         }
-
+        if (!isRole(entity.getRoleId())) {
+            return new ValidationResponse(false, "roleId must refer to a valid role");
+        }
```

Comment:
since roleId is already not null (point) at this point, how about just calling roleServiceClient.roleIdExists... here instead of having a private method?

Reply:
OK done that